woocommerce/woocommerce-ios
# [Free Trial] Adds StorePlanSynchronizer

Part of #9446 #9436 

# Why

This PR adds a new type called `StorePlanSynchronizer` which is responsible for fetching and sharing the current WPcom site plan information across multiple places, via de `ServiceLocator` accessor.

This is needed because we will need to display information about the site plan in 3 different places;
- Free trial banner
- Upgrades View
- Hub View (To be done in #9436)

So, In order to not duplicate the site plan syncing logic, this PR adds the `StorePlanSynchronizer` and integrates it in the Free Trial Banner and in the Upgrades View.

The next PRs will use the synchronizer to:

- Render plan name badge in the hub view.
- Show the free trial banner in all of the tab bars root views.

# Testing Steps

This PR does not change any existing behavior but make sure 
- The free trial banner appears on free trial sites.
- The free trial banner disappears after upgrading the plan from a free trial site.
- The Upgrades screen displays the correct site plan information.

---
